### PR TITLE
s4580286

### PR DESCRIPTION
**Pull request by shreya-ami**
Hi, 
This is a pull request to merge the contents of s4580286_rcnn into the topic_recognition/recognition branch. The problem attempted is COMP3710 Task 3: Detect lesions within the ISIC 2017/8 data set with the Mask RCNN.

Files: 

mask_rcnn_main.py - The main test driver script which implements the entire algorithm. It loads and trains the Mask R-CNN model and predicts the labels and masks of the test dataset. 

modules.py - Loads the pretrained Mask R-CNN model from torchvision models

dataset.py  - Loads and preproccesses images and masks into appropriately formatted tensors

train.py - Trains the model with the chosen hyperparameters and saves into a pytorch binary model file. 

predict.py - Evaluates and visualises accuracy and IoU of the test dataset

README.md - Provides more information on the model and implementation 

figures/ - Contains all figures used within README.md 

p.s. I was unsure as to why a new README.md was created inside the recognition folder. Please ignore this and use the README.md within the folder s4580286_rcnn/
Thanks,
Shreya Singh s45802867

### CODE REVIEW

**Comment by nathasha-naranpanawa** [comment]
**This is an initial inspection, no action is required at this point**

Difficulty: Normal

Readme:  OK
- not much information on data preprocessing
- **incorrect use of labels/images**(?)
- seem to have used segmentation maps as the ground truth itself instead of deriving the bounding boxes from them
- not sure why you are predicting lesions on binary masks (this is incorrect)
- some training loss and accuracy(mAP) plots shown

Commit messages:  OK

Code:
- No hyperparameter tuning or attempts to improve performance
- Minimal comments

Functionality/Performance:
- Classification accuracy is low, but IoU>80

**Comment by shreya-ami** [comment]
#### In reference to the comments above: 
> "seem to have used segmentation maps as the ground truth itself instead of deriving the bounding boxes from them"

Bounding box creation for ground truth are within dataset.py. [Link to code](https://github.com/Shreya-Personal/s4580286/blob/c4f0626011d0c26efcdebae3110a7bcc74f447ff/recognition/s4580286_rcnn/dataset.py#L62). 


>"not sure why you are predicting lesions on binary masks (this is incorrect)" 

The lesions are predicted from raw images. However, the readme shows the predicted mask and bounding box overlayed on the original image to show validity of both the segmentation and bounding box.  [Link to code](https://github.com/Shreya-Personal/s4580286/blob/c4f0626011d0c26efcdebae3110a7bcc74f447ff/recognition/s4580286_rcnn/predict.py#L123). 

>"No hyperparameter tuning or attempts to improve performance"

The hyperparameter tuning comes in the form of the learning rate scheduler within train.py.[ Link to code](https://github.com/Shreya-Personal/s4580286/blob/c4f0626011d0c26efcdebae3110a7bcc74f447ff/recognition/s4580286_rcnn/train.py#L74). 

Please let me know if this addresses the comments.

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation
Good spacing but minimal comments -1
Header blocks
## Recognition Problem 
Solves problem 
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage 
Module present
Commenting could be better -1
No Data leakage
Difficulty: Normal -5
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe good
Model/technical explanation
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No Feedback required
Request Description good
